test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -5711,5 +5711,232 @@ public void Dotnet_New_Template_Restore_Pack_Success(string template)
                 Assert.True(File.Exists(nupkgPath));
             }
         }
+
+        [Fact]
+        public async Task PackCommand_PackProject_PackageReference_PreReleaseDependency_Fails()
+        {
+            // Arrange
+            using (var pathContext = msbuildFixture.CreateSimpleTestPathContext())
+            {
+                var prereleaseDependencyName = "PreReleasePackageA";
+                var prereleaseDependencyVersion = "6.0.0-preview.3";
+                var prereleaseDependencyPackage = new SimpleTestPackageContext(prereleaseDependencyName, prereleaseDependencyVersion);
+                prereleaseDependencyPackage.Files.Clear();
+                prereleaseDependencyPackage.AddFile("_._");
+                await SimpleTestPackageUtility.CreatePackagesAsync(pathContext.PackageSource, prereleaseDependencyPackage);
+
+                var projectName = "ClassLibrary1";
+                string testDirectory = pathContext.WorkingDirectory;
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+                msbuildFixture.CreateDotnetNewProject(testDirectory, projectName);
+                string projectXml = $@"<Project Sdk=""Microsoft.NET.Sdk"">
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <Version>1.2.3</Version>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include=""{prereleaseDependencyName}"" Version=""{prereleaseDependencyVersion}""/>
+  </ItemGroup>
+</Project>";
+                File.WriteAllText(projectFile, projectXml);
+
+                // Act
+                msbuildFixture.RestoreProject(workingDirectory, projectName, string.Empty);
+                CommandRunnerResult result = msbuildFixture.PackProject(workingDirectory, projectName, $"/p:PackageOutputPath={workingDirectory}", validateSuccess: false);
```

Comment:
The test name implies its checking that something fails, but the test is just seeing if a package is created and if a warning is logged.  I would expect the test name to end in like "_SucceedsAndLogsWarning"?

Reply:
In general `pack` command generate nupkg file, but it's not creating one, so I'm assuming this is failure. I changed name to reflect log warning part.

Reply:
The test name says it fails, but it doesn't fail. 

In fact in line 5752

```cs
                Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
```

You're asserting success.

Reply:
Thank you for correcting, just addressed.